lib/modules/firestore/TransactionHandler.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * @flow
+ * Firestore Transaction representation wrapper
+ */
+import { getAppEventName, SharedEventEmitter } from '../../utils/events';
+import { getLogger } from '../../utils/log';
+import { getNativeModule } from '../../utils/native';
+import Transaction from './Transaction';
+import type Firestore from './';
+
+let transactionId = 0;
+
+/**
+ * Uses the push id generator to create a transaction id
+ * @returns {number}
+ * @private
+ */
+const generateTransactionId = (): number => transactionId++;
+
+export type TransactionMeta = {
+  id: number,
+  stack: Array<string>,
+  reject: null | Function,
+  resolve: null | Function,
+  transaction: Transaction,
+  updateFunction: (transaction: Transaction) => Promise<any>,
+};
+
+type TransactionEvent = {
+  id: number,
+  type: 'update' | 'error' | 'complete',
+  error: ?{ code: string, message: string },
+};
+
+/**
+ * @class TransactionHandler
+ */
+export default class TransactionHandler {
+  _firestore: Firestore;
+  _transactionListener: Function;
```

Comment:
Is there any reason to store this as we never unsubscribe from the listener?

Reply:
Hmm true good point, I just based it off the original stuff I did on RTDB transaction, guess we should remove storing on there as well tbh: https://github.com/invertase/react-native-firebase/blob/master/lib/modules/database/transaction.js#L24

Reply:
Yeah, may as well.
When I did the flow types for auth and firestore, I cleaned up a lot of this sort of stuff, but haven't got to database yet.

Reply:
Will remove on both, ta